src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,122 @@
+package duke;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.task.Todo;
+
+public class Storage {
+
+    private String filePath;
+
+    /**
+     * Constructor for Storage.
+     * @param filePath String of the file path leading to the data storage file
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+
+    }
+
+    /**
+     * Parses a line of input from the data storage, creates the corresponding task
+     * and adds it to the taskList.
+     *
+     * @param input String input from data storage
+     * @param taskList ArrayList of Tasks that contains the tasks
+     * @throws DukeException when invalid data exists in the data storage
+     */
+    private static void parseAndCreateTask(String input, ArrayList<Task> taskList) throws DukeException {
+        String[] inputArr = input.split(",");
+        Task currentTask;
+        switch (inputArr[0]) {
+        case "T" :

```

Comment:
For case statements, there shouldn't be a whitespace between "T" and :

Just gonna paste this for reference

```
switch (condition) {
case ABC:
    statements;
    // Fallthrough
case DEF:
    statements;
    break;
case XYZ:
    statements;
    break;
default:
    statements;
    break;
}
```